hydra-node/test/Hydra/HeadLogicSpec.hs

```diff
@@ -135,6 +135,28 @@ spec =
             getState
         getConfirmedSnapshot snapshotConfirmed `shouldBe` Just snapshot1
 
+      describe "ReqSn" $ do
+        it "prunes local txs in order" $ do
```

Comment:
In case anyone else is curious, this is the error you get if you revert the change to HeadLogic and run this test:

```
  test/Hydra/HeadLogicSpec.hs:157:24: 
  1) Hydra.HeadLogic, Coordinated Head Protocol, ReqSn, prunes local txs in order
       expected: [SimpleTx {txSimpleId = 2, txInputs = fromList [2], txOutputs = fromList [3]},SimpleTx {txSimpleId = 3, txInputs = fromList [3], txOutputs = fromList [4]}]
        but got: [SimpleTx {txSimpleId = 2, txInputs = fromList [2], txOutputs = fromList [3]},SimpleTx {txSimpleId = 1, txInputs = fromList [], txOutputs = fromList [2]}]
```